dim-web/src/routes/websocket.rs

```diff
@@ -47,23 +48,21 @@ where
     }
 }
 
-async fn ctrl_event_processor<A, T>(mut rx: mpsc::Receiver<CtrlEvent<A, T>>)
+async fn ctrl_event_processor<A, T>(mut rx: mpsc::UnboundedReceiver<CtrlEvent<A, T>>)
 where
     A: Hash + Eq + Clone,
     T: ToOwned<Owned = String> + Send,
 {
     let mut peers = HashMap::new();
     let mut discard = vec![];
 
-    loop {
+    while let Some(ev) = rx.recv().await {
         for addr in &discard {
             let _ = peers.remove(addr);
         }
 
         discard.clear();
 
-        let Some(ev) = rx.recv().await else { break };
```

Comment:
No. Looks like a sink.send was blocking enough to fill up the queue, causing subsequent sender's to freeze up and that back pressure travelled all the way to the axum handlers. The correct solution here is to tokio::spawn any portion code with an await that could block indefinitely and to not leak memory you have to wrap that in a timeout and drop the sink /peer from the map. This current solution only looks like it's working but will leak because we're not cleaning out the bad entries.